server/src/utils.ts

```diff
@@ -38,20 +38,23 @@ export let findProjectRootOfFile = (
   }
 };
 
-export let findBinaryFromProjectRoot = (
-  source: p.DocumentUri
+export let findBinaryDirPathFromProjectRoot = (
+  projectRootPath: p.DocumentUri // This must be a directory and not a file!
 ): null | p.DocumentUri => {
-  let dir = path.dirname(source);
-  let bscNativeReScriptPath = path.join(dir, c.nodeModulesBinDir);
+  let binaryDirPath = path.join(projectRootPath, c.nodeModulesBinDir);
+  let binaryPath = path.join(binaryDirPath, c.bscBinName);
```

Comment:
Is there a special reason for looking for `bsc` and not `rescript`? The latter seems more future-proof.

Reply:
Notice it's explicitly non-backwards-compatible with `bs-platform`. In the changelog.

Reply:
So `bsc` might in fact find the wrong dir by mistake. If `bs-platform` is around.

Reply:
This is also why this needs to be documented and put in the release notes.
So it's exactly clear where and how the binary is searched for.